lib/pure/asyncftpclient.nim

```diff
@@ -293,17 +299,20 @@ proc retrFile*(ftp: AsyncFtpClient, file, dest: string,
   ## The ``EvRetr`` event is passed to the specified ``handleEvent`` function
   ## when the download is finished. The event's ``filename`` field will be equal
   ## to ``file``.
-  var destFile = open(dest, mode = fmWrite)
   await ftp.pasv()
-  var reply = await ftp.send("RETR " & file.normalizePathSep)
-  assertReply reply, ["125", "150"]
-  if {'(', ')'} notin reply.string:
-    raise newException(ReplyError, "Reply has no file size.")
+  var reply = await ftp.send("size " & file.normalizePathSep)
```

Comment:
Why is this necessary? The ``SIZE`` command is part of [RFC 3659](https://tools.ietf.org/html/rfc3659), what if the server doesn't support it?

What should be done instead is not failing when the file size is not specified by ``RETR``'s reply, instead just download the file until EOF. As described by section 3.4 of https://tools.ietf.org/html/rfc959